File: app/src/main/java/io/github/sithengineer/motoqueiro/util/CompositeSubscriptionManager.java
package io.github.sithengineer.motoqueiro.util;

import rx.Subscription;
import rx.subscriptions.CompositeSubscription;

public class CompositeSubscriptionManager {

  private final CompositeSubscription subscriptions = new CompositeSubscription();

  public void add(Subscription subscription) {
    subscriptions.add(subscription);
  }

  public void clearAll() {
    if (subscriptions.hasSubscriptions() && !subscriptions.isUnsubscribed()) {
      subscriptions.unsubscribe();
    }
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/util/CompositeSubscriptionManager.kt
package io.github.sithengineer.motoqueiro.util

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

class CompositeSubscriptionManager {

  private val subscriptions = CompositeDisposable()

  fun add(subscription: Disposable) {
    subscriptions.add(subscription)
  }

  fun clearAll() {
    if (subscriptions.size() > 0 && !subscriptions.isDisposed) {
      subscriptions.dispose()
    }
  }
}
